pkg/awsvpc/parser.go

```diff
@@ -0,0 +1,344 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package awsvpc
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+
+	"github.com/np-guard/cloud-resource-collector/pkg/aws"
+	"github.com/np-guard/models/pkg/ipblock"
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/commonvpc"
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/logging"
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/vpcmodel"
+)
+
+type AWSresourcesContainer struct {
+	aws.ResourcesContainer
+}
+
+// parseResourcesFromFile returns aws.ResourcesContainer object, containing the configured resources structs
+// from the input JSON file
+func (rc *AWSresourcesContainer) ParseResourcesFromFile(fileName string) error {
+	inputConfigContent, err := os.ReadFile(fileName)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(inputConfigContent, &rc)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func mergeResourcesContainers(rc1, rc2 *AWSresourcesContainer) (*AWSresourcesContainer, error) {
+	if rc2 == nil && rc1 != nil {
+		return rc1, nil
+	}
+	if rc2 != nil && rc1 == nil {
+		return rc2, nil
+	}
+	if rc2 == nil && rc1 == nil {
+		return nil, fmt.Errorf("error merging input vpc resources files")
+	}
+	rc1.VpcsList = append(rc1.VpcsList, rc2.VpcsList...)
+	rc1.SubnetsList = append(rc1.SubnetsList, rc2.SubnetsList...)
+	rc1.NetworkACLsList = append(rc1.NetworkACLsList, rc2.NetworkACLsList...)
+	rc1.SecurityGroupsList = append(rc1.SecurityGroupsList, rc2.SecurityGroupsList...)
+	rc1.InternetGWList = append(rc1.InternetGWList, rc2.InternetGWList...)
+	rc1.InstancesList = append(rc1.InstancesList, rc2.InstancesList...)
+
+	return rc1, nil
+}
+
+func (rc *AWSresourcesContainer) VpcConfigsFromFiles(fileNames []string, vpcID, resourceGroup string, regions []string) (
+	*vpcmodel.MultipleVPCConfigs, error) {
+	for _, file := range fileNames {
+		mergedRC := &AWSresourcesContainer{}
+		err1 := mergedRC.ParseResourcesFromFile(file)
+		if err1 != nil {
+			return nil, fmt.Errorf("error parsing input vpc resources file: %w", err1)
+		}
+		rc, err1 = mergeResourcesContainers(mergedRC, rc)
+		if err1 != nil {
+			return nil, err1
+		}
+	}
+	vpcConfigs, err2 := rc.VPCConfigsFromResources(vpcID, resourceGroup, regions)
+	if err2 != nil {
+		return nil, fmt.Errorf("error generating cloud config from input vpc resources file: %w", err2)
+	}
+	return vpcConfigs, nil
+}
+
+func (rc *AWSresourcesContainer) filterByVpc(vpcID string) map[string]bool {
+	shouldSkipVpcIds := make(map[string]bool)
+	for _, vpc := range rc.VpcsList {
+		if vpcID != "" && *vpc.VpcId != vpcID {
```

Comment:
Please add an explaining comment (I really need it in order to understand)

Reply:
when the user provide a vpcID (other vpcs should be skipped) or when the user provide a region or resource-group (not supported in aws yet)

Reply:
please add to documentation

Reply:
added